core/controllers/blog_dashboard.py

```diff
@@ -174,24 +204,30 @@ def get(self, blog_post_id: str) -> None:
         if blog_post is None:
             raise self.PageNotFoundException(
                 'The blog post with the given id or url doesn\'t exist.')
-        user_settings = user_services.get_users_settings(
-            [blog_post.author_id], strict=False, include_marked_deleted=True)
-        username = user_settings[0].username
 
+        user_settings = user_services.get_user_settings(
+            blog_post.author_id, strict=False)
+        if user_settings:
+            profile_picture_data_url = user_settings.profile_picture_data_url
+        else:
+            profile_picture_data_url = None
+
+        author_details = blog_services.get_blog_author_details(
```

Comment:
Please handle the None case of `author_details`

Reply:
Removed optional[..] from `get_blog_author_details()`